core/src/main/java/com/datastax/oss/driver/internal/core/resolver/ResolverProvider.java

```diff
@@ -26,7 +31,25 @@ public static Resolver getResolver(Class<?> clazz) {
    *
    * @param resolverFactoryImpl new {@link Resolver} factory.
    */
-  public static void setDefaultResolverFactory(AbstractResolverFactory resolverFactoryImpl) {
+  public static synchronized void setDefaultResolverFactory(
```

Comment:
Same here, no need to use `synchronized` here by same reason.

Reply:
One thread can call `setDefaultResolverFactory`, go past the check if any resolver is in use (to line 41), then another thread can call `getResolver` and complete before first thread sets the resolver.

Reply:
`synchronized` does not solves it, because it makes particular method atomic, not both of them.
If you want to solve this case you need to use a lock.